apps/native-component-list/src/screens/BrightnessScreen.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { ScrollView, Text, View, Slider } from 'react-native';
+import * as Brightness from 'expo-brightness';
+import * as Permissions from 'expo-permissions';
+import HeadingText from '../components/HeadingText';
+import Button from '../components/Button';
+
+interface State {
+  initBrightness: { [type: string]: number };
+  sliderBrightness: { [type: string]: number };
+  systemBrightnessPermissionGranted: boolean;
+}
+
+const brightnessTypes: string[] = ['Brightness', 'SystemBrightness'];
+
+export default class BrightnessScreen extends React.Component<{}, State> {
+  static navigationOptions = {
+    title: 'Brightness',
+  };
+
+  readonly state: State = {
+    initBrightness: {},
+    sliderBrightness: {},
+    systemBrightnessPermissionGranted: true,
+  };
+
+  componentDidMount() {
+    Brightness.getBrightnessAsync().then(value => {
+      this.setState({ initBrightness: { ...this.state.initBrightness, Brightness: value } });
+    });
+
+    Brightness.getSystemBrightnessAsync().then(value => {
+      this.setState({ initBrightness: { ...this.state.initBrightness, SystemBrightness: value } });
+    });
+
+    Permissions.getAsync(Permissions.SYSTEM_BRIGHTNESS).then(result => {
+      if (result.status !== 'granted') {
+        this.setState({ systemBrightnessPermissionGranted: false });
+      }
+    });
+  }
+
+  render() {
+    let views = brightnessTypes.map(type => {
+      return (
+        <View key={type} style={{ padding: 20 }}>
+          <HeadingText>{type}</HeadingText>
+          {type === 'SystemBrightness' && (
+            <Button
+              title="Permissions.SYSTEM_BRIGHTNESS"
+              onPress={() => {
+                Permissions.askAsync(Permissions.SYSTEM_BRIGHTNESS).then(result => {
```

Comment:
Can you convert all those functions inside `render` to be class members? That would be much more clear I guess 🤔

Reply:
Thanks for the suggestions! I've fixed them 😄